### ISSUE TITLE
`no-background-only-scenario` rule produces false positive when scenarios are nested in Rules

### ISSUE THREAD

**Issue by Botsfloor** [gplint/gplint#1009]
The no-background-only-scenario rule incorrectly flags a Background as having "only one scenario" when all scenarios are organized under Rules, even when there are many
  scenarios that would inherit from the Background.

  **Expected Behavior:**
  The rule should count ALL scenarios that inherit from a Background, including those nested within Rules.

  **Actual Behavior:**
  The rule only counts direct child scenarios of the container with the Background, ignoring scenarios nested in Rules.

  **Root Cause:**
  In `src/rules/no-background-only-scenario.ts`, the logic only checks immediate children:
  `const scenariosInContainer = container.children.filter((c: any) => c.scenario).length;`

  This doesn't account for scenarios nested within Rules that would still inherit the Background.

  **Example:**
```
  Feature: Example feature with Background and Rules

    Background:
      Given some common setup step
      And another setup condition

    Rule: First group of related scenarios

      Scenario: First scenario in this rule
        When I perform some action
        Then I should see expected result

      Scenario: Second scenario in this rule
        When I perform different action
        Then I should see different result

    Rule: Second group of related scenarios

      Scenario: First scenario in second rule
        When I do something else
        Then I get another outcome

      Scenario: Second scenario in second rule
        When I try final action
        Then I see final result
```

  In this case, both scenarios use the Background, but the rule sees 0 direct scenarios and flags it as an error.

  **Suggested Fix:**
  The rule should recursively count all scenarios that would inherit from the Background, not just immediate children.
